conda/plugins/hookspec.py

```diff
@@ -410,11 +413,14 @@ def conda_request_headers(self) -> Iterable[[CondaRequestHeader]]:
 
 
            @plugins.hookimpl
-           def conda_request_headers():
-               yield plugins.CondaRequestHeader(
-                   name="Example-Header",
-                   description="This is an example HTTP header",
-                   value="example",
-                   hosts={"example.com", "sub.example.com"},
-               )
+           def conda_request_headers(method: str, url: conda.common.url.Url):
+               if url.scheme in ("https", "http") and url.netloc in {
+                   "example.com",
+                   "sub.example.com",
+               }:
+                   yield plugins.CondaRequestHeader(
+                       name="Example-Header",
+                       value="example",
+                   )
```

Comment:
I think this conditional yielding is something new in the plugin API that would be good to explore first a little, as a required first-level API, at least. To me, this smells like the type of API that might backfire since it would wholesale allow plugins to add headers without defining for what endpoints.

I thought we'd define `scheme`, `netloc` etc as attributes of the conda request header class, to be validated/checked against later in the process if the request location would match that. We could alternatively use `conda.common.url.Url` as the domain object and use a dunder comparison to decide if the session request location equals the request header URL. But, seeing how the `Url.__str__` is just using string interpolation for recreating the URL, makes me worried TBH, not even using `unparse` :-/

Reply:
> I think this conditional yielding is something new in the plugin API that would be good to explore first a little, as a required first-level API, at least.

FYI these hook args are opt-in only (https://pluggy.readthedocs.io/en/stable/#opt-in-arguments), dunno if that helps us much here but something interesting to keep in mind.